types/datum.go

```diff
@@ -1294,6 +1294,10 @@ func (d *Datum) convertToMysqlDecimal(sc *stmtctx.StatementContext, target *Fiel
 	if err == nil && err1 != nil {
 		err = err1
 	}
+	if err != nil {
+		ret.SetMysqlDecimal(dec)
```

Comment:
I think this can be removed?

Reply:
But I think this error can't be ignored.

Reply:
PTAL @breeswish @lysu

Reply:
I have added comment the this line. Do you think it is appropriate? PTAL @breeswish @lysu

Reply:
could you add a test to cover the code?

Reply:
L242 in `types/convert_test.go` can reach the code.

Reply:
when I remove your code here, the test can also pass...

Reply:
Do you run the test `TestConvertType`?

Reply:
Yes I did

Reply:
I have sent a screenshot to you. See slack.